eng/ApiCompatibility/PublicApiAnalyzer.props

```diff
@@ -0,0 +1,17 @@
+<Project>
+
+  <PropertyGroup Condition="'$(UsePublicApiAnalyzers)' == 'true'">
+    <!-- https://github.com/dotnet/roslyn-analyzers/blob/main/src/PublicApiAnalyzers/Microsoft.CodeAnalysis.PublicApiAnalyzers.md -->
+    <WarningsAsErrors>$(WarningsAsErrors);RS0016;RS0017;RS0022;RS0024;RS0025;RS0036;RS0037;RS0041;RS0048</WarningsAsErrors>
+
+    <!-- 
+      TODO: Enable the following:
+      * RS0026: Symbol 'XYZ' violates the backcompat requirement: 'Do not add multiple overloads with optional parameters'.
+        https://github.com/dotnet/extensions/issues/4167
+
+      * RS0027: 'XYZ' violates the backcompat requirement: 'API with optional parameter(s) should have the most parameters amongst its public overloads'.
+        https://github.com/dotnet/extensions/issues/4168
+    -->
+  </PropertyGroup>
```

Comment:
@mavasani @sharwell can you please give me a hand? I expect RS0026 and RS0027 not be reported, yet the [build is failing](https://dev.azure.com/dnceng-public/public/_build/results?buildId=335244&view=logs&j=774c6b69-2ab9-5dfa-262f-030e439e3fb3&t=336fa683-8901-5b1e-f931-6369c8775392&l=250) because of these. What am I missing? Thanks

Reply:
Why would you expect them to not be reported?

Reply:
📝 The WarningsCheck CI leg isn't publishing its build.binlog as an artifact, so it's difficult to diagnose the compiler input.s

Reply:
> Why would you expect them to not be reported?

Why wouldn't I? :) But I realise it was late when I asked the question, and my brain was fried at that point. The warning would be emitted regardless, the `WarningsAsErrors` turns the warning into an error.

How can I suppress the warnings though? I tried `<NoWarn>$(NoWarn);RS0026;RS0027</NoWarn>` but that didn't seem to work.

Reply:
@RussKie Both `NoWarn` and `WarningsNotAsErrors` should ensure these diagnostics are not flagged as errors. Are you able to extract out the compiler command line for the failing build and share it here?

Reply:
Looks like suppressing the warnings in the conditional group was the issue, once I moved the suppression in an unconditional group, it worked.